src/test/kotlin/com/exactpro/th2/check1/rule/AbstractCheckTaskTest.kt

```diff
@@ -51,6 +61,56 @@ abstract class AbstractCheckTaskTest {
         }
     }
 
+    protected fun extractEventBody(verificationEvent: Event): List<IBodyData> {
+        return jacksonObjectMapper()
+                .addMixIn(IBodyData::class.java, IBodyDataMixIn::class.java)
+                .readValue(verificationEvent.body.toByteArray())
+    }
+
+    protected fun assertVerification(verificationEvent: Event): Verification {
+        val verifications = extractEventBody(verificationEvent)
+        val verification = verifications.filterIsInstance<Verification>().firstOrNull()
+        assertNotNull(verification) { "Verification event does not contain the verification" }
+        return verification
+    }
+
+    protected fun assertVerificationEntries(
+            expectedVerificationEntries: Map<String, VerificationEntry>,
+            actualVerificationEntries: Map<String, VerificationEntry>?,
+            asserts: (VerificationEntry, VerificationEntry) -> Unit) {
+        assertNotNull(actualVerificationEntries) { "Actual verification entry is null" }
+        expectedVerificationEntries.forEach { (expectedFieldName, expectedVerificationEntry) ->
+            val actualVerificationEntry = actualVerificationEntries[expectedFieldName]
+            assertNotNull(actualVerificationEntry) {
+                "Actual verification entry does not contains field '${expectedFieldName}'"
+            }
+            asserts(expectedVerificationEntry, actualVerificationEntry)
+            if (expectedVerificationEntry.fields == null) {
+                return
```

Comment:
The `forEach` function is inline one. Calling `return` will cause exiting the whole loop instead of skipping a single iteration
```suggestion
                return@forEach
```